src/OpenTelemetry.Extensions.PersistentStorage.Abstractions/PersistentStorageAbstractionsEventSource.cs

```diff
@@ -0,0 +1,64 @@
+﻿// <copyright file="PersistentStorageAbstractionsEventSource.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Threading;
+
+namespace OpenTelemetry.Extensions.PersistentStorage
+{
+    [EventSource(Name = EventSourceName)]
+    internal sealed class PersistentStorageAbstractionsEventSource : EventSource
+    {
+        public static PersistentStorageAbstractionsEventSource Log = new PersistentStorageAbstractionsEventSource();
+        private const string EventSourceName = "OpenTelemetry-Extensions-PersistentStorage-Abstractions";
+
+        [NonEvent]
+        public void LogException(string message, Exception ex)
+        {
+            if (this.IsEnabled(EventLevel.Error, EventKeywords.All))
+            {
+                this.LogException(message, ToInvariantString(ex));
+            }
+        }
+
+        [Event(1, Message = "{0} : {1}", Level = EventLevel.Error)]
+        public void LogException(string message, string ex)
```

Comment:
I'm curious to know if having dedicated events for `PersistentBlob` and `PersistentBlobProvider` make it easier when troubleshooting?

Reply:
The event names will be different so, it will be more clear where the exception occurred. I will update it